client/src/LandingPages/SignupPage/SignupPage.jsx

```diff
@@ -3,62 +3,133 @@ import './SignupPage.css';
 import { Link } from 'react-router-dom';
 import LandingPageNav from '../components/LandingPageHeader/LandingPageNav';
 import { useState } from 'react';
+import FormInput from '../components/Forms/FormInput'
+import PasswordValidation from '../components/Forms/PasswordValidation';
+import { MdAlternateEmail } from "react-icons/md";
+import { TbPencilCheck } from "react-icons/tb";
+import { RiLockPasswordFill } from "react-icons/ri";
+import { FaUserPlus } from "react-icons/fa6";
+import * as authService from '../../services/authService';
 
-const SignupPage = () => {
+const SignupPage = ({ setUser }) => {
   const [inputValue, setInputValue] = useState({
+    firstName: '',
+    lastName: '',
     email: '',
-    password: ''
+    password: '',
+    confirmPassword: '',
   });
 
+  const [errorMsg, setErrorMsg] = useState('')
+  const [successMsg, setSuccessMsg] = useState('')
+
+
+  // Add more form fields as needed here:
+  const formFields = [
+    {
+      name: 'firstName',
+      label: 'First Name',
+      type: 'text',
+      id: "signup-firstName",
+      htmlFor: "signup-firstName",
+      errorMessage: "First name needs to be 2 to 20 characters long and shouldn't contain special characters.",
+      pattern: "^[A-Za-z]{2,20}$",
+      icon: <TbPencilCheck />,
+      required: true,
+    },
+    {
+      name: 'lastName',
+      label: 'Last Name',
+      type: 'text',
+      id: "signup-lastName",
+      htmlFor: "signup-lastName",
+      errorMessage: "Last name needs to be 2 to 20 characters long and shouldn't contain special characters.",
+      pattern: "^[A-Za-z]{2,20}$",
+      icon: <TbPencilCheck />,
+      required: true,
+    },
+    {
+      name: 'username',
+      label: 'Username',
+      type: 'text',
+      id: "signup-username",
+      htmlFor: "signup-username",
+      errorMessage: "Username should be 3-20 characters. Only letters, numbers and underscores are allowed.",
+      pattern: "^[A-Za-z0-9_]{3,20}$",
+      icon: <FaUserPlus />,
+      required: true,
+    },
+    {
+      name: 'email',
+      label: 'Email',
+      type: 'email',
+      id: "signup-email",
+      htmlFor: "signup-email",
+      errorMessage: "Please use a valid email address.",
+      icon: <MdAlternateEmail />,
+      required: true,
+    },
+    {
```

Comment:
Also, if sign-up fails, along with producing an error message for the users, ideally the password and confirmed password fields should be cleared out if a user's sign up attempt fails after they click on submit.